### PR TITLE
[NNCF] Enable FP16 in mmdetection

### PR DESCRIPTION
**Pull request by daniil-lyakhov**
With improvement introduced in https://github.com/openvinotoolkit/nncf/pull/1279 models could be optimized in FP16 precision. This change allows to use FP16 during optimization stage

### CODE REVIEW

File: external/mmdetection/detection_tasks/apis/detection/nncf_task.py [hunk 1]
```diff
@@ -230,12 +230,6 @@ def optimize(
         self._is_training = True
         self._model.train()
 
-        fp16 = training_config.get("fp16", None)
```
Comment (alexsu52) [inline external/mmdetection/detection_tasks/apis/detection/nncf_task.py:233]: Should NNCF version be updated for this changes?

**Review by alexsu52 — CHANGES_REQUESTED** [review]

File: external/mmdetection/detection_tasks/apis/detection/nncf_task.py (hunk 1 above)
Comment (daniil-lyakhov) [reply external/mmdetection/detection_tasks/apis/detection/nncf_task.py:233]: Yes, it should

Comment (alexsu52) [reply external/mmdetection/detection_tasks/apis/detection/nncf_task.py:233]: Let's update it here

**Comment by sovrasov** [comment]
Closed as outdated

**Pull request closed without merging**